goblin/driver/connection.py

```diff
@@ -35,17 +32,14 @@ def __init__(self, response_queue, loop, conn, username, password,
             raise StopAsyncIteration
 
     async def fetch_data(self):
-        if not self._response_queue.empty():
-            message = self._response_queue.get_nowait()
-        else:
-            self._loop.create_task(self._conn.get_data())
-            message = await self._response_queue.get()
-        return message
+        if self._done:
+            return None
 
```

Comment:
Because if the response has already been marked as done, this will short-circuit the `await self._response_queue.get()` which would wait forever since there are no more messages.